### PR TITLE
Newton-Raphson state vector scaling

### PR DESCRIPTION
**Pull request by geofjamg**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: src/main/java/com/powsybl/openloadflow/ac/nr/LineSearchStateVectorRescaler.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class LineSearchStateVectorRescaler implements StateVectorRescaler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LineSearchStateVectorRescaler.class);
+
+    private static final int MAX_ITERATION = 10;
+    private static final int STEP_FOLD = 4;
```
Comment (Hadrien-Godard) [inline src/main/java/com/powsybl/openloadflow/ac/nr/LineSearchStateVectorRescaler.java:25]: mu = 0.25 seems very conservative to me. Maybe something smoother like 0.75 (STEP_FOLD = 4/3) is more appropriate.

Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/ac/nr/LineSearchStateVectorRescaler.java]: But in that case we should increase the number of max iterations?

Comment (Hadrien-Godard) [reply src/main/java/com/powsybl/openloadflow/ac/nr/LineSearchStateVectorRescaler.java]: 0.75^10 = 0.056... so I think it is sufficient, don't you agree ?

Comment (geofjamg) [reply src/main/java/com/powsybl/openloadflow/ac/nr/LineSearchStateVectorRescaler.java]: Yes, let's try with 0.75

**Review by jeandemanged — APPROVED** [review]

File: src/main/java/com/powsybl/openloadflow/ac/nr/MaxVoltageChangeStateVectorScaling.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            switch (variable.getType()) {
+                case BUS_V:
+                    if (absValueChange > maxDv) {
+                        stepSize = Math.min(stepSize, maxDv / absValueChange);
+                        vCutCount++;
+                    }
+                    break;
+                case BUS_PHI:
+                    if (absValueChange > maxDphi) {
+                        stepSize = Math.min(stepSize, maxDv / absValueChange);
```
Comment (geofjamg) [inline src/main/java/com/powsybl/openloadflow/ac/nr/MaxVoltageChangeStateVectorScaling.java:68]: maxDphi?

**Review by Hadrien-Godard — APPROVED** [review]

File: src/main/java/com/powsybl/openloadflow/ac/nr/MaxVoltageChangeStateVectorScaling.java (hunk 2 above)
Comment (jeandemanged) [reply src/main/java/com/powsybl/openloadflow/ac/nr/MaxVoltageChangeStateVectorScaling.java:68]: oopsie

Comment (jeandemanged) [reply src/main/java/com/powsybl/openloadflow/ac/nr/MaxVoltageChangeStateVectorScaling.java]: thank you @Hadrien-Godard for the fix

**Pull request merged**